src/payment/strategies/googlepay/googlepay-stripe-upe-initializer.ts

```diff
@@ -0,0 +1,107 @@
+import { round } from 'lodash';
+
+import { Checkout } from '../../../checkout';
+import { InvalidArgumentError } from '../../../common/error/errors';
+import PaymentMethod from '../../payment-method';
+
+import { BillingAddressFormat, GooglePaymentData, GooglePayInitializer, GooglePayPaymentDataRequestV2, TokenizePayload } from './googlepay';
+
+export default class GooglePayStripeUPEInitializer implements GooglePayInitializer {
+    initialize(
+        checkout: Checkout,
+        paymentMethod: PaymentMethod,
+        hasShippingAddress: boolean
+    ): Promise<GooglePayPaymentDataRequestV2> {
+        return Promise.resolve(this._getGooglePayPaymentDataRequest(
+            checkout,
+            paymentMethod,
+            hasShippingAddress
+        ));
+    }
+
+    teardown(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    parseResponse(paymentData: GooglePaymentData): Promise<TokenizePayload> {
+        try {
+            const payload = JSON.parse(paymentData.paymentMethodData.tokenizationData.token);
+
+            return Promise.resolve({
```

Comment:
Can we have some typeguard for payload to check if it actually has `id` and `type` values?

Same for `card` below?